src/overlays/actors/ovl_En_Siofuki/z_en_siofuki.c

```diff
@@ -21,27 +30,286 @@ const ActorInit En_Siofuki_InitVars = {
     (ActorFunc)EnSiofuki_Update,
     (ActorFunc)EnSiofuki_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Siofuki/EnSiofuki_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Siofuki/EnSiofuki_Destroy.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+extern Gfx D_06000B70[];
+extern UNK_TYPE D_06000D78;
+
+void EnSiofuki_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnSiofuki* this = THIS;
+    s32 type;
+    ColHeader* colHeader = NULL;
+    s32 pad;
+
+    if ((thisx->room == 10) && Flags_GetSwitch(globalCtx, 0x1E)) {
+        Actor_Kill(thisx);
+        return;
+    }
+
+    Actor_ProcessInitChain(thisx, sInitChain);
+    DynaPolyInfo_SetActorMove(&this->dyna, DPM_PLAYER);
+    DynaPolyInfo_Alloc(&D_06000D78, &colHeader);
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, thisx, colHeader);
+    this->sfxFlags |= 1;
+
+    type = ((u16)thisx->params >> 0xC) & 0xF;
+    if (!((type == 0) || (type == 1))) {
+        Actor_Kill(thisx);
+        return;
+    }
+
+    this->initPosY = thisx->posRot.pos.y;
+    this->unk_174 = 35.0f;
+    this->unk_170 = -6058.0f + this->unk_174;
+
+    if (thisx->shape.rot.x != 0) {
+        this->maxHeight = thisx->shape.rot.x * 40.0f;
+        this->currentHeight = this->maxHeight;
```

Comment:
```suggestion
        this->currentHeight = this->maxHeight = thisx->shape.rot.x * 40.0f;
```

Reply:
The two separate lines are required to match.